bootstrap/viaproxy/src/main/java/org/geysermc/geyser/platform/viaproxy/GeyserViaProxyPlugin.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright (c) 2019-2023 GeyserMC. http://geysermc.org
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ * @author GeyserMC
+ * @link https://github.com/GeyserMC/Geyser
+ */
+
+package org.geysermc.geyser.platform.viaproxy;
+
+import net.lenni0451.lambdaevents.EventHandler;
+import net.raphimc.viaproxy.ViaProxy;
+import net.raphimc.viaproxy.plugins.PluginManager;
+import net.raphimc.viaproxy.plugins.ViaProxyPlugin;
+import net.raphimc.viaproxy.plugins.events.ConsoleCommandEvent;
+import net.raphimc.viaproxy.plugins.events.ProxyStartEvent;
+import net.raphimc.viaproxy.plugins.events.ShouldVerifyOnlineModeEvent;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.geysermc.geyser.GeyserImpl;
+import org.geysermc.geyser.session.GeyserSession;
+import org.geysermc.geyser.text.GeyserLocale;
+
+import java.io.File;
+import java.util.UUID;
+
+public class GeyserViaProxyPlugin extends ViaProxyPlugin {
+
+    public static final Logger LOGGER = LogManager.getLogger("Geyser");
+    public static final File ROOT_FOLDER = new File(PluginManager.PLUGINS_DIR, "Geyser");
+
+    private GeyserViaProxyBootstrap bootstrap;
+
+    @Override
+    public void onEnable() {
+        ROOT_FOLDER.mkdirs();
+
+        this.bootstrap = new GeyserViaProxyBootstrap(LOGGER, ROOT_FOLDER);
+        GeyserLocale.init(this.bootstrap);
+        this.bootstrap.onEnable();
+
+        ViaProxy.EVENT_MANAGER.register(this);
+    }
+
+    @EventHandler
+    private void onConsoleCommand(final ConsoleCommandEvent event) {
+        if (event.getCommand().equals("geyser") || event.getCommand().equals("/geyser")) {
```

Comment:
This check would break all Geyser extension provided commands. I'd recommend to implement a similar logic as Geyser has already for Geyser-Standalone to check if a command should be forwarded:

Here's the check that's only ran on standalone & ensures that geyser/extensions commands are not sent to the proxy:
https://github.com/GeyserMC/Geyser/blob/d63a70daa9604193d78824b6fb8c2ad3f9923b14/core/src/main/java/org/geysermc/geyser/translator/protocol/bedrock/BedrockCommandRequestTranslator.java#L42-L43

The check itself is done with the runCommand method of the GeyserCommandManager:
https://github.com/GeyserMC/Geyser/blob/d63a70daa9604193d78824b6fb8c2ad3f9923b14/core/src/main/java/org/geysermc/geyser/command/GeyserCommandManager.java#L148-L193

Reply:
That won't like that for ViaProxy since ViaProxy has its own commands and always running `GeyserImpl.getInstance().commandManager().runCommand` results in the "Invalid command" message from Geyser being sent for regular ViaProxy commands

Reply:
Not sure why it shouldn't work for Viaproxy?
The runCommand() function returns a boolean - either true if it's a geyser internal command (which wouldn't be passed to the java server/viaproxy), or also true if it's a command by a geyser extension, but false if Geyser doesn't know the command (aka, it's not a geyser cmd nor provided by an extension.

See https://github.com/GeyserMC/Geyser/blob/d63a70daa9604193d78824b6fb8c2ad3f9923b14/core%2Fsrc%2Fmain%2Fjava%2Forg%2Fgeysermc%2Fgeyser%2Fcommand%2FGeyserCommandManager.java#L157-L159 - I think you can just cancel the event when that returns true (?)

Reply:
The runCommand method always sends a message into the console if the command is unknown to Geyser ![image](https://github.com/GeyserMC/Geyser/assets/50594595/2838b8ae-111c-4aad-95d3-2ef64ffda916)

Reply:
it should never get there if it's not a geyser or extension command, the first return that I've linked takes care of that